src/main/webapp/js/tests/AuthButton.test.js

```diff
@@ -0,0 +1,84 @@
+import {
+  AuthButton,
+  USER_SIGNUP_URL,
+} from '../common/react/components/AuthButton.js';
+import '@testing-library/jest-dom/extend-expect';
+import {render, screen} from '@testing-library/react';
+
+/**
+ * Mock auth data representing when a user is authorized and has created a
+ * profile.
+ */
+const mockAuthData1 = {
```

Comment:
We can give these constants more descriptive names like `AUTHORIZED_WITH_PROFILE_AUTH_DATA`

Reply:
I agree, that definitely makes it more readable. Will change this now.